.github/workflows/release-azure-pipelines.yml

```diff
@@ -93,40 +93,268 @@ extends:
               targetPath: dist/${{ config.runtime }}
               artifactName: azureauth-${{ parameters.version }}-${{ config.runtime }}
           steps:
-            - checkout: self
-            - task: UseDotNet@2
-              displayName: Use .NET Core sdk 6.x
-              inputs:
-                version: 6.x
+          - checkout: self
+          - task: UseDotNet@2
+            displayName: Use .NET Core sdk 6.x
+            inputs:
+              version: 6.x
+          - task: NuGetToolInstaller@0
+            displayName: Use NuGet 6.x
+            inputs:
+              versionSpec: 6.x
+          - task: DotNetCoreCLI@2
+            displayName: Install dependencies
+            inputs:
+              command: restore
+              feedsToUse: select
+              vstsFeed: $(vstsFeedId)
+              includeNuGetOrg: false
+              arguments: --runtime ${{ config.runtime }}
+          - task: DotNetCoreCLI@2
+            displayName: Test
+            inputs:
+              command: test
+              arguments: --configuration release --no-restore
+          
+          - task: DotNetCoreCLI@2
+            displayName: Build artifacts
+            env:
+              ADO_TOKEN: $(System.AccessToken)
+            inputs:
+              command: publish
+              projects: src/AzureAuth/AzureAuth.csproj
+              arguments: -p:Version=${{ parameters.version }} --configuration release --self-contained true --runtime ${{ config.runtime }} --output dist/${{ config.runtime }}
+              publishWebProjects: false
+              zipAfterPublish: false
+              modifyOutputPath: true

```

Comment:
Does this task generate the artifacts to the `$(Build.ArtifactStagingDirectory)` directory? And that's how the artifacts are being used in line 146?

Reply:
That's almost correct. But instead, it generates artifacts to `dist/{{config.runtime}}` and we mention that path in the `templateContext` output `targetpath`. The artifacts mentioned in output are then uploaded (this time from `$(Build.ArtifactStagingDirectory)`)so it can be downloaded by signing stage. Similarly for signing stage where signed artifacts are uploaded at the end of stage for packaging stage.

Reply:
Awesome!